src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -190,11 +256,43 @@ void BaseCouplingScheme::advance()
 
   PRECICE_ASSERT(_couplingMode != Undefined);
 
+  double relativeDt                 = -1;
+  bool   usesFirstParticipantMethod = (_timeWindowSize == -1);
+
+  if (isImplicitCouplingScheme()) {
+    // store data from writeDataContext in buffer, when advance is called.
+    PRECICE_ASSERT(_computedTimeWindowPart > 0);
+    if (not usesFirstParticipantMethod) {
+      relativeDt = _computedTimeWindowPart / _timeWindowSize;
+      PRECICE_ASSERT(math::smallerEquals(relativeDt, 1.0, 10e-9), relativeDt, _computedTimeWindowPart, _timeWindowSize);
+      PRECICE_ASSERT(relativeDt > 0, relativeDt, _computedTimeWindowPart, _timeWindowSize);
+      if (relativeDt > 1.0) {
+        PRECICE_ASSERT(math::smallerEquals(relativeDt, 1.0, 10e-9), relativeDt);
```

Comment:
the 1e-9 seems arbitrary here :grin:
what are you trying to do?

Reply:
I removed a large part of this code and it still works. I assume this is a left-over from the "use a double as dict key"-times and the code has become unnecessary. See fb4d0f85d5de99f9b844bf68aad84ee7c47c7122.